x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SlmHealthIndicatorService.java

```diff
@@ -87,25 +108,83 @@ public HealthIndicatorResult calculate(boolean explain) {
             return createIndicator(
                 YELLOW,
                 "Snapshot Lifecycle Management is not running",
-                createDetails(explain, slmMetadata),
+                createDetails(explain, Collections.emptyList(), slmMetadata),
                 impacts,
                 List.of(SLM_NOT_RUNNING)
             );
         } else {
+            List<SnapshotLifecyclePolicyMetadata> unhealthyPolicies = slmMetadata.getSnapshotConfigurations()
+                .values()
+                .stream()
+                .filter(this::snapshotFailuresExceedWarningCount)
+                .toList();
+
+            if (unhealthyPolicies.size() > 0) {
+                List<HealthIndicatorImpact> impacts = Collections.singletonList(
+                    new HealthIndicatorImpact(
+                        2,
+                        "Some automated snapshots have not had a successful execution recently. Indices restored from affected "
+                            + "snapshots may not contain recent changes.",
+                        List.of(ImpactArea.BACKUP)
+                    )
+                );
+
+                return createIndicator(
+                    YELLOW,
+                    "Encountered [" + unhealthyPolicies.size() + "] unhealthy snapshot lifecycle management policies.",
+                    createDetails(explain, unhealthyPolicies, slmMetadata),
+                    impacts,
+                    List.of(
+                        new UserAction(
+                            ACTION_CHECK_RECENTLY_FAILED_SNAPSHOTS,
+                            unhealthyPolicies.stream().map(SnapshotLifecyclePolicyMetadata::getName).toList()
+                        )
+                    )
+                );
+            }
+
             return createIndicator(
                 GREEN,
                 "Snapshot Lifecycle Management is running",
-                createDetails(explain, slmMetadata),
+                createDetails(explain, Collections.emptyList(), slmMetadata),
                 Collections.emptyList(),
                 Collections.emptyList()
             );
         }
     }
 
-    private static HealthIndicatorDetails createDetails(boolean explain, SnapshotLifecycleMetadata metadata) {
+    private boolean snapshotFailuresExceedWarningCount(SnapshotLifecyclePolicyMetadata policyMetadata) {
```

Comment:
Should we have this package-visible, static (needs another parameter) and unit tested ?